### DISCUSSION TITLE
Only launch the server when there's a connection to the databse?

### DISCUSSION THREAD

**Discussion by kissingerbrian** [C-I-T-CLUB/CIT_CLUB_OFFICIAL_WEB_APP_-BACKEND#19] [General]
@PaymasterMax proposed a very brilliant idea during our discussion.
there idea is; The app should only start when the connection to the database was established successfully otherwise the app should thro an error and fail to start.

**Comment by kissingerbrian** [comment]
> @PaymasterMax proposed a very brilliant idea during our discussion. there idea is; The app should only start when the connection to the database was established successfully otherwise the app should thro an error and fail to start.

My honest opinion:
I don't agree with this approach.
This approach number:
 **1. Will be very expensive in the long run.**
     Let's say at the start, the connection to the database was established successfully, and during apping, 
     a databse connection might die and this would the entire application would fail and will be restart 
     because it can only be up when a database connection is up. 
    
     When a connection is lost, the server/app should not die but should try re-connecting to the database 
     while up and 
     it will save on the cost of waiting for the entire to start after each and every successful connection.
 
2. Database connection is a promise that is invoked and the code procedes to run other code as the database connection request is push to event queues awaiting for resolve or reject response. Once a responce is received it can be made available for the application. Because the database connection does not affect the application launching.

**Comment by kissingerbrian** [comment]
> @PaymasterMax proposed a very brilliant idea during our discussion. there idea is; The app should only start when the connection to the database was established successfully otherwise the app should thro an error and fail to start.

```
Application server started and listening on http://localhost:1880
Mongoose connected to the database
Conneted to the database successfully
Mongoose disconnected from the database
Mongoose connected to the database
Mongoose disconnected from the database
Mongoose connected to the database
Mongoose disconnected from the database
Mongoose connected to the database
Mongoose disconnected from the database
Mongoose connected to the database
Mongoose disconnected from the database
Mongoose connected to the database
Mongoose disconnected from the database
Mongoose connected to the database
```

Look at that snippet it means that our app would have restarted all those times the database was disconnected.

But for the implementation I have done, the server will keep re-connecting everytime the database is disconnected

**Comment by PaymasterMax** [comment]
No, the application will only restart once. Not so many times, the app.js
has the app.listen which is an event loop. It will only execute once unless
the server is manually restarted. If the dB connection fails mid procēss,
the app won't restart, you can test it out in your local environment and
see.

**Reply by kissingerbrian** [reply]
ok
